### PR TITLE
Bug 1182464 db auto starring

### PR DESCRIPTION
**Pull request by maurodoglio**
[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/853)

### CODE REVIEW

File: treeherder/log_parser/management/commands/store_error_summary.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from django.core.management.base import BaseCommand, CommandError
+from mozlog import reader
+
+from treeherder.model.derived import JobsModel
+from treeherder.model.models import Failure, Repository
+
+
+class Command(BaseCommand):
+    args = '<log_url>, <job_guid>, <repository>'
+    help = 'Download, parse and store the given failure summary log.'
```
Comment (edmorley) [inline treeherder/log_parser/management/commands/store_error_summary.py]: We currently have a task to download and ingest a log (https://github.com/mozilla/treeherder/blob/master/treeherder/etl/management/commands/test_parse_log.py) - is it worth bringing that over into the new treeherder/log_parser/management/commands/ directory too?

**Comment by edmorley** [comment]
Review status: 0 of 14 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

<sup>**[treeherder/log_parser/management/commands/store_error_summary.py, line 13 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS5D7UFJeEXsjvrZbT)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/log_parser/management/commands/store_error_summary.py#L13)):</sup>
We currently have a task to download and ingest a log (https://github.com/mozilla/treeherder/blob/master/treeherder/etl/management/commands/test_parse_log.py) - is it worth bringing that over into the new treeherder/log_parser/management/commands/ directory too?

---

<sup>**[treeherder/log_parser/tasks.py, line 68 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS5jdCVXYOVyMTAIOA)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/log_parser/tasks.py#L68)):</sup>
I think this should be `store_error_summary.retry()`... otherwise we may have some interesting intermittent behaviour :-)

---

<sup>**[treeherder/model/migrations/0001_initial.py, line 307 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6xDu2v9xI-bLmEyG)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/migrations/0001_initial.py#L307)):</sup>
s/assignements/assignments/g

---

<sup>**[treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl, line 703 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6DgcgNqqIZMMBI8n)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl#L703)):</sup>
There are tabs here.

---

<sup>**[treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl, line 706 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6MtrxW8naS6zwf-2)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl#L706)):</sup>
Are these ALTER statements because of Django generating the table?
Confused why the properties aren't listed on the original CREATE TABLE declaration of the fields?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)

**Comment by maurodoglio** [comment]
Review status: 0 of 14 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

<sup>**[treeherder/log_parser/management/commands/store_error_summary.py, line 13 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS5D7UFJeEXsjvrZbT)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/log_parser/management/commands/store_error_summary.py#L13)):</sup>
Yeah that would make sense

---

<sup>**[treeherder/log_parser/tasks.py, line 68 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS5jdCVXYOVyMTAIOA)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/log_parser/tasks.py#L68)):</sup>
whoops :stuck_out_tongue_closed_eyes: 

---

<sup>**[treeherder/model/migrations/0001_initial.py, line 307 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6xDu2v9xI-bLmEyG)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/migrations/0001_initial.py#L307)):</sup>
I got rid of those fields :smile_cat: 

---

<sup>**[treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl, line 0 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6MtrxW8naS6zwf-2)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl#L0)):</sup>
Yeah they were auto generated. I got rid of them.

---

<sup>**[treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl, line 703 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6DgcgNqqIZMMBI8n)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl#L703)):</sup>
It should be ok now

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)

**Comment by jgraham** [comment]
Review status: 0 of 16 files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

<sup>**[treeherder/model/models.py, line 558 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwbyWYy-NwqRc8hQCes)** ([raw file](https://github.com/mozilla/treeherder/blob/08b42bb7c7c014f57f0710c0b0abffdda517b573/treeherder/model/models.py#L558)):</sup>
This should allow null (I think the same applies to the following columns as well).

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)

**Comment by maurodoglio** [comment]
Review status: 0 of 16 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

<sup>**[treeherder/log_parser/management/commands/store_error_summary.py, line 0 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS5D7UFJeEXsjvrZbT)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/log_parser/management/commands/store_error_summary.py#L0)):</sup>
Done.

---

<sup>**[treeherder/log_parser/tasks.py, line 0 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS5jdCVXYOVyMTAIOA)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/log_parser/tasks.py#L0)):</sup>
Done.

---

<sup>**[treeherder/model/migrations/0001_initial.py, line 0 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6xDu2v9xI-bLmEyG)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/migrations/0001_initial.py#L0)):</sup>
Done.

---

<sup>**[treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl, line 0 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6MtrxW8naS6zwf-2)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl#L0)):</sup>
Done.

---

<sup>**[treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl, line 0 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6DgcgNqqIZMMBI8n)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl#L0)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)

**Comment by edmorley** [comment]
Reviewed 10 of 14 files at r1, 3 of 6 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

<sup>**[bin/run_celery_worker_log_parser, line 26 \[r3\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-Jwy0WF3dHtdhqWGTE2A)** ([raw file](https://github.com/mozilla/treeherder/blob/c51aeb449a9fbe9b2c07e3288c678320d169a827/bin/run_celery_worker_log_parser#L26)):</sup>
nit: There's an additional space before `store_error_summary` but not between the other items in the list.

---

<sup>**[treeherder/log_parser/utils.py, line 39 \[r3\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-Jwy2JvquNCGVWPIdGBT)** ([raw file](https://github.com/mozilla/treeherder/blob/c51aeb449a9fbe9b2c07e3288c678320d169a827/treeherder/log_parser/utils.py#L39)):</sup>
This method of scheduling the tasks is quite different from how the existing `mozlog_json` logs are scheduled (they are added as log URLs). Is there a reason for not using the same method for both? (Changing `mozlog_json` to use this way is just as fine, so long as we're consistent :-)). 

eg:
https://github.com/mozilla/treeherder/blob/1c00ccfcc74275f3ff7366c6150614bb065a8dc2/treeherder/etl/buildapi.py#L196-L200
https://github.com/mozilla/treeherder/blob/3418d2e5ccf1f2d82334dc276e9233c2c11a77d0/treeherder/model/derived/jobs.py#L1711-L1714

---

<sup>**[treeherder/model/models.py, line 23 \[r3\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-Jwy1SibW1bPUAozifdx)** ([raw file](https://github.com/mozilla/treeherder/blob/c51aeb449a9fbe9b2c07e3288c678320d169a827/treeherder/model/models.py#L23)):</sup>
The isort run put relative path imports separate from absolute path imports - so we need an additional newline before this line to be consistent with that.

---

<sup>**[treeherder/model/models.py, line 573 \[r3\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-Jwy1f8cNFovALNlHjCB)** ([raw file](https://github.com/mozilla/treeherder/blob/c51aeb449a9fbe9b2c07e3288c678320d169a827/treeherder/model/models.py#L573)):</sup>
Thank you for changing these model names after our chat in the office - they read a lot more clearly now :-)

---

<sup>**[treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl, line 704 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6MtrxW8naS6zwf-2)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl#L704)):</sup>
But they're still here? (re: "Confused why the properties aren't listed on the original CREATE TABLE declaration of the fields?")

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)

**Comment by jgraham** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

<sup>**[treeherder/model/models.py, line 573 \[r3\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-Jwy1f8cNFovALNlHjCB)** ([raw file](https://github.com/mozilla/treeherder/blob/c51aeb449a9fbe9b2c07e3288c678320d169a827/treeherder/model/models.py#L573)):</sup>
FWIW I am not super convinced about the new names after coding against them for a while; I think that Failure is misleading for something that isn't necessarily a test failure (or even an atomic thing; one overall error could be the result of a number of lines); for consistency with errorsummary it could probably be called ErrorLine or something. If this changed FailureMatch would have to have a corresponding change. More importantly, KnownIntermittentFailure is too long. The "Known" prefix at least seems unnecessary. 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)

**Comment by maurodoglio** [comment]
Review status: 11 of 16 files reviewed at latest revision, 4 unresolved discussions, some commit checks broke.

---

<sup>**[treeherder/log_parser/utils.py, line 39 \[r3\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-Jwy2JvquNCGVWPIdGBT)** ([raw file](https://github.com/mozilla/treeherder/blob/c51aeb449a9fbe9b2c07e3288c678320d169a827/treeherder/log_parser/utils.py#L39)):</sup>
The way the mozlog_json is detected makes it only work forjobs coming from buildbot. An interesting question is whether we want to keep parsing mozlog_json even if we have a superset of the content available in the error_summary

---

<sup>**[treeherder/model/models.py, line 573 \[r3\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-Jwy1f8cNFovALNlHjCB)** ([raw file](https://github.com/mozilla/treeherder/blob/c51aeb449a9fbe9b2c07e3288c678320d169a827/treeherder/model/models.py#L573)):</sup>
I'll change those names according to jgraham's feedback 

---

<sup>**[treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl, line 704 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JwS6MtrxW8naS6zwf-2)** ([raw file](https://github.com/mozilla/treeherder/blob/20b54adee51699c56bdc6f720dd6d57b189fc0f1/treeherder/model/sql/template_schema/treeherder_reference_1.sql.tmpl#L704)):</sup>
I was referring to the ALTER statements to add new columns (like ). I don't know why the migration framework is spitting out these.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)

**Comment by jgraham** [comment]
Review status: 5 of 16 files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

<sup>**[treeherder/model/models.py, line 548 \[r5\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JxeK5A5mR8bkjpDGBDV)** ([raw file](https://github.com/mozilla/treeherder/blob/689be80f6496743264690b8d5f67123569e14da6/treeherder/model/models.py#L548)):</sup>
I think the max length here is *probably* OK but only because it happens to be enforced by length restrictions on Windows path names. That said, I can imagine some edge cases in which it would break (e.g. a test with a URL and query string as part of the name)

---

<sup>**[treeherder/model/models.py, line 549 \[r5\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JxeKQz6YjX99TUY1BzP)** ([raw file](https://github.com/mozilla/treeherder/blob/689be80f6496743264690b8d5f67123569e14da6/treeherder/model/models.py#L549)):</sup>
A max length here is also problematic, even moreso than the above in fact because there is nothing limiting the length of a subtest name. Looking at web-platform-tests we have tests with up to 395 characters in the subtest name, I imagine other suites might also have long names here.

---

<sup>**[treeherder/model/models.py, line 552 \[r5\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JxjoWYWJDjIgKNPE_zU)** ([raw file](https://github.com/mozilla/treeherder/blob/689be80f6496743264690b8d5f67123569e14da6/treeherder/model/models.py#L552)):</sup>
This certainly needs to be a TextField(blank=True, null=True)

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)

**Comment by maurodoglio** [comment]
Review status: 5 of 16 files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

<sup>**[treeherder/model/models.py, line 548 \[r5\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JxeK5A5mR8bkjpDGBDV)** ([raw file](https://github.com/mozilla/treeherder/blob/689be80f6496743264690b8d5f67123569e14da6/treeherder/model/models.py#L548)):</sup>
Done.

---

<sup>**[treeherder/model/models.py, line 549 \[r5\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JxeKQz6YjX99TUY1BzP)** ([raw file](https://github.com/mozilla/treeherder/blob/689be80f6496743264690b8d5f67123569e14da6/treeherder/model/models.py#L549)):</sup>
Done.

---

<sup>**[treeherder/model/models.py, line 552 \[r5\]](https://reviewable.io:443/reviews/mozilla/treeherder/853#-JxjoWYWJDjIgKNPE_zU)** ([raw file](https://github.com/mozilla/treeherder/blob/689be80f6496743264690b8d5f67123569e14da6/treeherder/model/models.py#L552)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)

**Comment by jgraham** [comment]
Reviewed 1 of 5 files at r6.
Review status: 6 of 16 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/853)
